src/Microsoft.AspNetCore.Mvc.Razor/Compilation/CompiledViewManfiest.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Compilation
+{
+    public static class CompiledViewManfiest
+    {
+        public static readonly string PrecompiledViewsAssemblySuffix = ".PrecompiledViews";
+
+        public static TManifest LoadManifest<TManifest>(AssemblyPart assemblyPart, string typeName)
```

Comment:
My thought here is that this doesn't really need to be generic. This won't work with all types, it's only two specific ones.